### ISSUE TITLE
`indent-binary-ops`: False positive with parens (Even example of correct code fails)

### ISSUE THREAD

**Issue by kku39** [eslint-stylistic/eslint-stylistic#530]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Steps to reproduction

1. Open https://eslint.style/rules/default/indent-binary-ops#examples
  - I have checked the page of v2.6.2 and v2.7.2, and the both pages reproduced.
2. See "Examples of **correct** code for this rule:" (The following code)

```js
/*eslint @stylistic/indent-binary-ops: ["error", 2]*/

if (a
  && b
  && c
  && (d
    || e
    || f
  )
) {
  foo()
}
```

### Actual Behavior

```
    7:1   error    Expected indentation of 2 spaces                     @stylistic/indent-binary-ops
```

![image](https://github.com/user-attachments/assets/dad1c9bb-6ca5-40b0-a792-13604fd0da65)

### Expected Behavior

No error.

### Reproduction

https://eslint.style/rules/default/indent-binary-ops#examples

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by Brainshaker95** [comment]
I have also noticed some additional cases that aren't handled that well by this rule. These might be related to this issue so I'm leaving them here.  
Let me know if I should open a separate issue for those.

```js
/*eslint @stylistic/indent-binary-ops: ["error", 2]*/

// --- Example 1--- //

// No error
const a = () => b
  || c
  ? d
  : e;
  
// Error
const a = (
  p,
) => b
  || c // Expected indentation of 0 spaces
  ? d
  : e;

// --- Example 2 --- //

// No error
const a= b
  + c;

// Error
const a= {
  p: b
    + c, // Expected indentation of 2 spaces
};
```

**Comment by Brainshaker95** [comment]
@antfu Nice, thank you!
